Update run_test_btn.tsx for validation feedback

Eamonn-OL opened the pull request:
At the moment, the run button greys out and informs the user to fix the above fields, but it's not clear which fields it's referring to.

This change brings the "run" button in line with the "Create" button - the button is no longer disabled, however clicking the run button will still give you highlights to correct.

Eamonn-OL committed:
Update run_test_btn.tsx for validation feedback

At the moment, the run button greys out and informs the user to fix the above fields, but it's not clear which fields it's referring to.

This change brings the "run" button in line with the "Create" button - despite being disabled, clicking the run button will still give you highlights to correct.

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/303479)
* Commit: 25836bfdab0d4e84fc7f4b399bcbb67bc23a0e3b

### Failed CI Steps
* [Linting](https://buildkite.com/elastic/kibana-pull-request/builds/303479#01971d0c-ad3f-459f-ba1c-a8e944ed3ac6)

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/303479","number":303479,"commit":"25836bfdab0d4e84fc7f4b399bcbb67bc23a0e3b"}],"number":303479}
buildkite-pr-comment-->

Eamonn-OL commented:
This may need some enhancement based on the linting failure:

`error Visible, non-interactive elements with click handlers must have at least one keyboard listener jsx-a11y/click-events-have-key-events`

This was intentional, however I see the accessibility concern here. This may be better handled by using an `aria-disabled` type instead, however I'm not sure if that's supported in EUI?

Eamonn-OL committed:
Update run_test_btn.tsx

Changing approach based on discussions - enables the button similar to how the "create monitor" button works, but prompts for validation before allowing run.

Eamonn-OL committed:
Merge branch 'main' into Improve-Synthetics-Validation-Creation-Form

Eamonn-OL committed:
Update run_test_btn.tsx

Adjustments for linting failures as well as to reduce code redundancy based on how the create button is implemented.

Eamonn-OL committed:
Merge branch 'main' into Improve-Synthetics-Validation-Creation-Form

Eamonn-OL committed:
Update run_test_btn.tsx

Further linting cleanup

Eamonn-OL committed:
Merge branch 'main' into Improve-Synthetics-Validation-Creation-Form

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/309570)
* Commit: 91476632b404220a001572786d2c3e12231f30a1

### Failed CI Steps
* [Quick Checks](https://buildkite.com/elastic/kibana-pull-request/builds/309570#01977f07-0c1e-4ce5-83e2-a85e001dc0dc)

### History
* :broken_heart: [Build #309530](https://buildkite.com/elastic/kibana-pull-request/builds/309530) failed d00c3e08d402243afa147204a24e293e583b13b4
* :broken_heart: [Build #309480](https://buildkite.com/elastic/kibana-pull-request/builds/309480) failed c2b73545b5286a36493791e9eaee69cb7298236e
* :broken_heart: [Build #303479](https://buildkite.com/elastic/kibana-pull-request/builds/303479) failed 25836bfdab0d4e84fc7f4b399bcbb67bc23a0e3b

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309570","number":309570,"commit":"91476632b404220a001572786d2c3e12231f30a1"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309530","number":309530,"commit":"d00c3e08d402243afa147204a24e293e583b13b4"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/309480","number":309480,"commit":"c2b73545b5286a36493791e9eaee69cb7298236e"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/303479","number":303479,"commit":"25836bfdab0d4e84fc7f4b399bcbb67bc23a0e3b"}],"number":309570}
buildkite-pr-comment-->

Eamonn-OL committed:
Merge branch 'main' into Improve-Synthetics-Validation-Creation-Form

Eamonn-OL committed:
Removing unused i18n

Eamonn-OL committed:
Merge branch 'main' into Improve-Synthetics-Validation-Creation-Form

Eamonn-OL committed:
Update x-pack/solutions/observability/plugins/synthetics/public/apps/synthetics/components/monitor_add_edit/form/run_test_btn.tsx

Co-authored-by: Francesco Fagnani <fagnani.francesco@gmail.com>
